src/Test/Utilities/Portable/Compilation/IRuntimeEnvironment.cs

```diff
@@ -27,20 +27,13 @@ internal static IRuntimeEnvironment Create(IEnumerable<ModuleData> additionalDep
 
         private static IRuntimeEnvironmentFactory GetFactoryImplementation()
         {
-            string assemblyName;
-            string typeName;
-            if (CoreClrShim.IsRunningOnCoreClr)
-            {
-                assemblyName = "Roslyn.Test.Utilities.CoreClr";
-                typeName = "Microsoft.CodeAnalysis.Test.Utilities.CodeRuntime.CoreCLRRuntimeEnvironmentFactory";
-            }
-            else
-            {
-                assemblyName = "Roslyn.Test.Utilities.Desktop";
-                typeName = "Microsoft.CodeAnalysis.Test.Utilities.CodeRuntime.DesktopRuntimeEnvironmentFactory";
-            }
-
```

Comment:
Why is the runtime check not better? Adding #ifdef now requires this code to be built twice, even if we fix the deployment issue for test projects.